src/stem.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+var util = require("./util");
+
+
+/**
+ * Stem
+ *
+ * A class that creates and manages the "insert widget here" buttons a.k.a "stems"
+ * that appear beside an empty block level element.
+ */
+
+function Stem(blockElem, onClick) {
+  util.assert(!blockElem.$stem);
+
+  this.blockElem = blockElem;
+  this.blockElem.$stem = this;
+
+  this.stemButton = this.createDom();
+  this.reposition();
+
+  // Handle click event
+  this.stemButton.addEventListener( 'click', function() {
+    onClick();
+  }, false);
+
+}
+
+// Static "idompotent" function to retrive or create a Stem
+Stem.getOrCreate = function(blockElem, onClick) {
+  return blockElem.$stem || new Stem(blockElem, onClick);
+}
+
+
+// Hide the stem
+Stem.prototype.destroy = function() {
+  // Delete back reference
+  delete this.blockElem.$stem;
```

Comment:
or set to null
or don't have line at all (this kind of thing avoided memory leaks in IE6, not sure if it's an issue these days)
delete the two comments in this function, they don't add any information that isn't obvious